assets/apps/dashboard/src/utils/module-observer.js

```diff
@@ -0,0 +1,72 @@
+import { subscribe, select } from '@wordpress/data';
+
+/**
+ * Append New Sub Menu Page.
+ *
+ * The page will be added after the parent container of the linkSelector.
+ *
+ * @param {string} linkSelector The link selector to determine the position of the added submenu page.
+ * @param {Object} subMenuData The Sub Menu data to be added.
+ * @return {HTMLAnchorElement} Custom Layouts menu page.
+ */
+function appendNewSubMenuPage(linkSelector, subMenuData) {
+	// Create and configure a new anchor element
+	const link = document.createElement('a');
+	link.setAttribute('href', subMenuData?.linkSubMenu);
+	link.textContent = subMenuData?.labelSubMenu;
+
+	const linkToAppendAfter = document.querySelector(linkSelector);
+	if (!linkToAppendAfter) {
+		return;
+	}
+
+	// Append the after the Customizer menu item.
+	const listNode = linkToAppendAfter.parentNode;
+	const container = document.createElement('li');
+	container.appendChild(link);
+	listNode.parentNode.insertBefore(container, listNode.nextSibling);
+
+	return link;
+}
+
+/**
+ * Auto Sub Menu Pages on Admin Dashboard for Pro Modules.
+ */
+function autoHideModuleSubMenuPages() {
+	let clLinkElem = document.querySelector(
+		'a[href*="edit.php?post_type=neve_custom_layouts"]'
+	);
+
+	subscribe(() => {
+		/**
+		 * Auto hide Custom Layouts submenu page link based on module status.
+		 */
+		const isModuleEnabled =
+			select('neve-dashboard').getModuleStatus('custom_layouts');
+
+		if (isModuleEnabled && !clLinkElem) {
+			clLinkElem = appendNewSubMenuPage(
+				'.wp-submenu a[href*="customize.php"]',
```

Comment:
There are 2 menu items that point to Customizer, one under `Neve`, the other is under `Appearance`.
You need to ensure you target the proper one, also you need to account for the White label module that can replace the theme Name inside the menu.